petab/v2/core.py

```diff
@@ -0,0 +1,711 @@
+"""Types around the PEtab object model."""
+from __future__ import annotations
+
+from enum import Enum
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import sympy as sp
+from pydantic import (
+    BaseModel,
+    Field,
+    ValidationInfo,
+    field_validator,
+    model_validator,
+)
+
+from ..v1.lint import is_valid_identifier
+from ..v1.math import sympify_petab
+from . import C
+
+__all__ = [
+    "Observable",
+    "ObservablesTable",
+    "ObservableTransformation",
+    "NoiseDistribution",
+    "Change",
+    "ChangeSet",
+    "ConditionsTable",
+    "OperationType",
+    "ExperimentPeriod",
+    "Experiment",
+    "ExperimentsTable",
+    "Measurement",
+    "MeasurementTable",
+    "Mapping",
+    "MappingTable",
+    "Parameter",
+    "ParameterScale",
+    "ParameterTable",
+]
+
+
+class ObservableTransformation(str, Enum):
+    """Observable transformation types.
+
+    Observable transformations as used in the PEtab observables table.
+    """
+
+    LIN = C.LIN
+    LOG = C.LOG
+    LOG10 = C.LOG10
+
+
+class ParameterScale(str, Enum):
+    """Parameter scales.
+
+    Parameter scales as used in the PEtab parameters table.
+    """
+
+    LIN = C.LIN
+    LOG = C.LOG
+    LOG10 = C.LOG10
+
+
+class NoiseDistribution(str, Enum):
+    """Noise distribution types.
+
+    Noise distributions as used in the PEtab observables table.
+    """
+
+    NORMAL = C.NORMAL
+    LAPLACE = C.LAPLACE
+
+
+class ObjectivePriorType(str, Enum):
+    """Objective prior types.
+
+    Objective prior types as used in the PEtab parameters table.
+    """
+
+    NORMAL = C.NORMAL
+    LAPLACE = C.LAPLACE
+    UNIFORM = C.UNIFORM
+    LOG_NORMAL = C.LOG_NORMAL
+    LOG_LAPLACE = C.LOG_LAPLACE
+    PARAMETER_SCALE_NORMAL = C.PARAMETER_SCALE_NORMAL
+    PARAMETER_SCALE_LAPLACE = C.PARAMETER_SCALE_LAPLACE
+    PARAMETER_SCALE_UNIFORM = C.PARAMETER_SCALE_UNIFORM
+
+
+assert set(C.PRIOR_TYPES) == {e.value for e in ObjectivePriorType}, (
+    "ObjectivePriorType enum does not match C.PRIOR_TYPES: "
+    f"{set(C.PRIOR_TYPES)} vs { {e.value for e in ObjectivePriorType} }"
+)
+
+
+class InitializationPriorType(str, Enum):
+    """Initialization prior types.
+
+    Initialization prior types as used in the PEtab parameters table.
+    """
+
+    NORMAL = C.NORMAL
+    LAPLACE = C.LAPLACE
+    UNIFORM = C.UNIFORM
+    LOG_NORMAL = C.LOG_NORMAL
+    LOG_LAPLACE = C.LOG_LAPLACE
+    PARAMETER_SCALE_NORMAL = C.PARAMETER_SCALE_NORMAL
+    PARAMETER_SCALE_LAPLACE = C.PARAMETER_SCALE_LAPLACE
+    PARAMETER_SCALE_UNIFORM = C.PARAMETER_SCALE_UNIFORM
```

Comment:
Looks like this belongs in v2/C.py, as possibly also remove `C.PRIOR_TYPES` (or deprecate and define it in terms of `ObjectivePriorType`. Is there a reason to have both `ObjectivePriorType` and `InitializationPriorType`?

Reply:
I don't know if they will always be the same. Not strong opinion.

Reply:
Alright, while they are the same then I think this makes sense
```python
class PriorType:
    ...

ObjectivePriorType = PriorType
InitializationPriorType = PriorType
```